File: Phoenix/lib/bd_starter/bd/resources/subscriptions.ex
defmodule BdStarter.Bd.Resources.Subscriptions do
  @moduledoc "Subscription plans + the Stripe checkout hand-off for one."

  alias BdStarter.Bd.Client

  def list(client), do: Client.get(client, "subscriptions")
  def get(client, id), do: Client.get(client, "subscriptions/#{enc(id)}")

  def checkout(client, id, urls),
    do: Client.post(client, "subscriptions/#{enc(id)}/checkout", urls)

  defp enc(v), do: URI.encode(to_string(v), &URI.char_unreserved?/1)
end

File: Laravel/app/Bd/Resources/Subscriptions.php
<?php

namespace App\Bd\Resources;

use App\Bd\Client;

/** Subscription plans + the Stripe checkout hand-off for one. */
class Subscriptions
{
    public function __construct(private readonly Client $client)
    {
    }

    /** @return array<string, mixed> */
    public function list(): array
    {
        return $this->client->get('subscriptions');
    }

    /** @return array<string, mixed> */
    public function get(string $id): array
    {
        return $this->client->get('subscriptions/'.rawurlencode($id));
    }

    /**
     * @param  array<string, mixed>  $input  successUrl / cancelUrl / customerEmail
     * @return array<string, mixed>
     */
    public function checkout(string $id, array $input): array
    {
        return $this->client->post('subscriptions/'.rawurlencode($id).'/checkout', $input);
    }
}
